packages/grid/src/vaadin-grid-data-provider-mixin.js

```diff
@@ -397,26 +397,33 @@ export const DataProviderMixin = (superClass) =>
             cache.size = items.length;
           }
 
-          const currentItems = Array.from(this.$.items.children).map((row) => row._item);
-
           // Populate the cache with new items
           items.forEach((item, itemsIndex) => {
             const itemIndex = page * this.pageSize + itemsIndex;
             cache.items[itemIndex] = item;
-            if (this._isExpanded(item) && currentItems.indexOf(item) > -1) {
-              // Force synchronous data request for expanded item sub-cache
-              cache.ensureSubCacheForScaledIndex(itemIndex);
+          });
+
+          // With the new items added, update the cache size and the grid's effective size
+          this._cache.updateSize();
+          this._effectiveSize = this._cache.effectiveSize;
+
+          // After updating the cache, check if some of the expanded items should have sub-caches loaded
+          this._getVisibleRows().forEach((row) => {
```

Comment:
As far as I see, there is a `_effectiveSize` observer that basically does the same. It requests items by calling `virtualizer.update()` which eventually results in `grid._getItem(...)`. That takes place when `this._hasData` is `true` which is almost always the case except after you have cleared the cache. So, this `ensureSubCacheForScaledIndex` should be already invoked there.

Reply:
👍 Thx, I'll see if there's room for improvement there also